perfzero/lib/cloud_manager.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/python
+
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Helper script to create, query and stop machine in GCP."""
+
+from __future__ import print_function
+
+import argparse
+import getpass
+import logging
+import subprocess
+import sys
+
+
+INSTANCE_NAME_PREFIX = 'perfzero-dev-'
+
+
+def run_command(cmd, is_from_user=False):
+  """Runs list of command and throw error if return code is non-zero.
+
+  Args:
+    cmd: Command to execute
+    is_from_user: If true, log the command and the command output in INFO level.
+                  Otherwise, log these in the DEBUG level.
+
+  Returns:
+    a string representing the command output
+
+  Raises:
+    Exception: raised when the command execution has non-zero exit code
+  """
+  if is_from_user:
+    logging.info('Executing command: %s\n', cmd)
+  else:
+    logging.debug('Executing command: %s\n', cmd)
+
+  stdout = ''
+  p = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                       stderr=subprocess.STDOUT, shell=True)
+  exit_code = None
+  while exit_code is None:
+    exit_code = p.poll()
+    line = p.stdout.readline().decode('utf-8').strip()
+    if not line:
+      continue
+
+    if is_from_user:
+      logging.info(line)
+    else:
+      logging.debug(line)
+    stdout = stdout + line + '\n'
+
+  if exit_code and is_from_user:
+    sys.exit(exit_code)
+  elif exit_code:
+    raise Exception('Command:\n{}\nfailed with output:\n{}'.format(cmd, stdout))
+
+  return stdout
+
+
+def get_instance_name(username):
+  return INSTANCE_NAME_PREFIX + username
+
+
+def get_machine_type(accelerator_count):
+  """Get machine type string based on the accelerator count.
+
+  Args:
+    accelerator_count: accelerator count
+
+  Returns:
+    a string recognized by the gcloud utility as the machine type
+  """
+  if accelerator_count == 0:
```

Comment:
The numbers by default would be
1 gpu   8 cpu
2 gpu  16 cpu
4    32 CPU
8  64 CPU
There is a 96 CPU version

Reply:
I have updated the machine_type logic to be the following:

  - Use the user-specified machine_type if it is not None
  - Otherwise, use the standard type with cpu_count = 8 x accelerator_count
    if user-specified accelerator_count > 0
  - Otherwise, use the standard type with 8 cpu